nbclient/client.py

```diff
@@ -411,6 +411,8 @@ def setup_kernel(self, **kwargs):
         When control returns from the yield it stops the client's zmq channels, and shuts
         down the kernel.
         """
+        reset_kc = kwargs.pop('reset_kc', False)
```

Comment:
I'd rename this to `cleanup_kc` and change `def setup_kernel(self, **kwargs):` to `def setup_kernel(self, cleanup_kc=False, **kwargs):`

Reply:
Ahh I see you just copied the async kernel implementation -- we should probably clean that up as well to make it clearer how it works, but I can also see the argument for keeping the current convention to avoid breaking people.